### PR TITLE
Error loading details shows error alert

### PR DESCRIPTION
**Pull request by FuriousLlama**
Improved handling of the details request. 
Added an alert in case there was an error loading medication statements details
[AB#7719](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/7719)

Fixed the headers not being passed into the get request. 
Fixed an issue where invalid Keywords where being allowed. 
Fixed unittests for login
[AB#7723](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/7723)

### CODE REVIEW

File: Apps/Medication/src/Controllers/MedicationController.cs [hunk 1]
```diff
@@ -90,6 +91,7 @@ public RequestResult<MedicationResult> GetMedication(string drugIdentifier)
 
             RequestResult<Dictionary<string, MedicationResult>> result = new RequestResult<Dictionary<string, MedicationResult>>()
             {
+                ResultStatus = Common.Constants.ResultType.Sucess,
```
Comment (TiagoGraf89) [inline Apps/Medication/src/Controllers/MedicationController.cs]: typo Sucess > Success

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_229)

File: Apps/Medication/src/Controllers/MedicationController.cs (hunk 1 above)
Comment (FuriousLlama) [reply Apps/Medication/src/Controllers/MedicationController.cs]: good catch. Fixed it, it propagated everywhere

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_229)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_229)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_229)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_229)

**Pull request merged**